src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,295 @@
+is retrieved from the PAM conversation and written to the stdin of the
+forked passkey child process(Similar to ``get_p11_child_write_buffer()``).
+Check the passkey_child return code, return PAM_SUCCESS or failure based on the
+result and call ``pam_reply()``. If the credential is discoverable, then the
+passkey_child also prints the ``userId``, and the PAM responder has to
+compare it with the one provided by the LDAP server. If they match, then it can
+return PAM_SUCCESS.
+
+Prompting implementation
+************************
+The prompting is handled by the PAM responder. For that purpose, the passkey
+related authtokens is added in ``src/util/authtok.c``,
+``sss_authtok_set_passkey_pin()`` and ``sss_authtok_get_passkey_pin()``.
+
+In ``src/sss_client/pam_sss.c`` add a ``prompt_passkey()`` option which takes
+the prompt as an argument to call ``do_pam_conversation()``. This sets the
+type SSS_AUTHTOK_TYPE_PASSKEY_PIN.
+
+In the PAM responder, ``pam_set_passkey_prompting_options()`` is added to
+``src/responder/pam/pam_prompting_config.c`` to check and handle the PAM
+prompting passkey configuration options.
+
+``pc_list_add_passkey()`` copies the interactive prompt message to
+``pc->data.passkey.prompt_inter`` and the touch prompt message to
+``pc->data.passkey.prompt_touch`` and set type PC_TYPE_PASSKEY.
+
+If interactive or touch options are set to false, then these values are set to
+an empty string. If pam_sss reads an empty string for these prompts it does not
+include them in the PAM conversation. This is done because we can't
+fallback to a default prompt message in this situation, we need to skip the
+interactive or touch prompt entirely.
+
+Add the prompt message to the data buffer response back to pam_sss by adding
+PC_TYPE_PASSKEY to case statements in ``pam_get_response_prompt_config()`` and
+``pc_list_from_response()``. 
+
+In ``prompt_passkey()``, 1 to 3 messages are provided to the PAM conversation
+function. The first “Enter PIN:” PAM message is always created. If interactive
+and/or touch prompts are enabled in the prompt configuration then those
+messages are added to the pam message array and provided to the pam
+conversation.
+
+.. _configuration-options:
+
+Configuration options
+*********************
+"pam_passkey_auth" enables the passkey device authentication.
+
+"passkey_verification" is added to the SSSD configuration options. It is
+similar to the “certificate_verification” option for the p11_child, as it
+contains the parameters needed to tune the passkey_child helper
+process. The list of parameters and their meaning:
+
+* user-verification: if set to true, requires user verification (i.e. PIN,
+  password) during authentication. If set to false, does not request user
+  verification during authentication. The default is that the key itself
+  decides what to do.
+
+* debug-libfido2: flag to print libfido2 library messages. Defaults to false.
```

Comment:
Should this really be inside passkey_verification option?  Wouldn't a standalone option be better?

Reply:
I'd like to keep the configuration behaviour as similar to the `p11_child` as possible. Changing it to a stand-alone option would do the contrary.

Reply:
I'm sorry, but this is debugging, it does not have anything to do with verification so it does not belong there. What similarities are you referring to? I don't see any debug value in `certificate_verification`.

Reply:
I meant, that all the configuration parameters for `p11_child` are defined in `certificate_verification`, and I'd like to do the same for passkey. It's true though that this is a debug parameter. Are you suggesting to create a new passkey section and move it to its own configuration option?

Reply:
You can create passkey_debug_libfido2 inside [pam].

Reply:
Done. Justin will take of changing the implementation to match this change.